### PR TITLE
Add Rust CPD

### PR DESCRIPTION
**Pull request by juliapaluch**
## Describe the PR

This PR adds CPD support for Rust.

## Ready?

- [x] Added unit tests for fixed bug/feature
- [x] Passing all unit tests
- [x] Complete build `./mvnw clean verify` passes (checked automatically by github actions)
- [x] Added (in-code) documentation (if needed)

### CODE REVIEW

File: pmd-rust/src/main/java/net/sourceforge/pmd/lang/rust/RustLanguageModule.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+import net.sourceforge.pmd.lang.LanguagePropertyBundle;
+import net.sourceforge.pmd.lang.LanguageRegistry;
+import net.sourceforge.pmd.lang.rust.cpd.RustCpdLexer;
+import net.sourceforge.pmd.lang.impl.CpdOnlyLanguageModuleBase;
+
+public class RustLanguageModule extends CpdOnlyLanguageModuleBase {
+    private static final String ID = "rust";
+
+    public RustLanguageModule() {
+        super(LanguageMetadata.withId(ID).name("Rust").extensions("rust"));
```
Comment (adangel) [inline pmd-rust/src/main/java/net/sourceforge/pmd/lang/rust/RustLanguageModule.java:17]:
```suggestion
        super(LanguageMetadata.withId(ID).name("Rust").extensions("rs"));
```

Comment (juliapaluch) [reply pmd-rust/src/main/java/net/sourceforge/pmd/lang/rust/RustLanguageModule.java]: thank you! addressed in da524115c0b943256a1466c4d0145b0fc49b1a8f

**Comment by juliapaluch** [comment]
good point! I added a README to the directory where the grammar lives. I also deleted `RustParser.g4` because I'm assuming that's unnecessary if we're just tokenizing. Please correct me if I'm wrong and I'll add it back!

File: pmd-rust/src/main/java/net/sourceforge/pmd/lang/rust/ast/RustLexer.java [hunk 2]
```diff
@@ -0,0 +1,1233 @@
...
+import org.antlr.v4.runtime.TokenStream;
+import org.antlr.v4.runtime.*;
+import org.antlr.v4.runtime.atn.*;
+import org.antlr.v4.runtime.dfa.DFA;
+import org.antlr.v4.runtime.misc.*;
+
+import net.sourceforge.pmd.lang.rust.ast.RustLexerBase;
+
+@SuppressWarnings({"all", "warnings", "unchecked", "unused", "cast", "CheckReturnValue", "this-escape"})
+public class RustLexer extends RustLexerBase {
```
Comment (jsotuyod) [inline pmd-rust/src/main/java/net/sourceforge/pmd/lang/rust/ast/RustLexer.java:18]: why is this file here? if it's generated from the grammar during build (under `build/generated-sources/antlr4`) there is no need to have it under `src/main/java`…

File: pmd-rust/src/main/java/net/sourceforge/pmd/lang/rust/ast/RustLexerBase.java [hunk 3]
```diff
@@ -0,0 +1,84 @@
+// CHECKSTYLE:OFF
+package net.sourceforge.pmd.lang.rust.ast;
+
+import org.antlr.v4.runtime.*;
+
+public abstract class RustLexerBase extends Lexer{
```
Comment (jsotuyod) [inline pmd-rust/src/main/java/net/sourceforge/pmd/lang/rust/ast/RustLexerBase.java:6]: ditto

File: pmd-rust/src/main/java/net/sourceforge/pmd/lang/rust/ast/RustLexer.java (hunk 2 above)
Comment (juliapaluch) [reply pmd-rust/src/main/java/net/sourceforge/pmd/lang/rust/ast/RustLexer.java]: oops! that's my bad. I inlined the custom functions from the ANTLR grammar in `RustLexer.g4` instead and added a note about the modification to the README in 0f58dae1f25ae3c6b72f6dc1e070d86599c3e716. thank you! :)

**Comment by pmd-test** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="true">No regression tested rules have been changed.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Review by adangel — APPROVED** [review]
Thanks!
